### PR TITLE
8330540: Rename the enum type CompileCommand to CompileCommandEnum

### PR DESCRIPTION
**Pull request by iklam**
`CompileCommand` is used both as a enum type ([compilerOracle.hpp](https://github.com/openjdk/jdk/blob/064628471b83616b4463baa78618d1b7a66d0c7c/src/hotspot/share/compiler/compilerOracle.hpp#L104)), and a global variable ([compiler_globals.hpp](https://github.com/openjdk/jdk/blob/064628471b83616b4463baa78618d1b7a66d0c7c/src/hotspot/share/compiler/compiler_globals.hpp#L304)).

This makes very awkward to the enum type -- we are forced to use `enum CompileCommand` in the source code whenever a type is needed:

This simple c++ file illustrates the problem:
```
enum class CompileCommand { a, b, c };
void foo(CompileCommand x) {}
char* CompileCommand; // can no longer use "CompileCommand" as a type
void good(enum CompileCommand x) {}
void bad(CompileCommand x) {}

$ g++ -c ~/enum.cpp
/home/iklam/enum.cpp:5:6: error: variable or field ?bad? declared void
    5 | void bad(CompileCommand x) {}
```

The fix is to rename the enum type to `CompileCommandEnum`.

This also makes it possible to forward-declare `CompileCommandEnum` (see vmEnum.hpp) without including compilerOracle.hpp. This improves HotSpot build time by reducing the number of .o files that include compilerOracle.hpp from 456 to 16.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330540](https://bugs.openjdk.org/browse/JDK-8330540): Rename the enum type CompileCommand to CompileCommandEnum (**Enhancement** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Dean Long](https://openjdk.org/census#dlong) (@dean-long - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18829/head:pull/18829` \
`$ git checkout pull/18829`

Update a local copy of the PR: \
`$ git checkout pull/18829` \
`$ git pull https://git.openjdk.org/jdk.git pull/18829/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18829`

View PR using the GUI difftool: \
`$ git pr show -t 18829`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18829.diff">https://git.openjdk.org/jdk/pull/18829.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18829#issuecomment-2062447200)

### CODE REVIEW

File: src/hotspot/share/jvmci/jvmciCompilerToVM.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@
 #include "code/scopeDesc.hpp"
 #include "compiler/compileBroker.hpp"
 #include "compiler/compilerEvent.hpp"
+#include "compiler/compilerOracle.hpp"
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciCompilerToVM.cpp:34]: Why you need it here?

Comment (iklam) [reply src/hotspot/share/jvmci/jvmciCompilerToVM.cpp:34]: It's needed by this:
```
jvmciCompilerToVM.cpp:571:21: error: 'CompilerOracle' has not been declared
  571 |   return !Inline || CompilerOracle::should_not_inline(method) || method->dont_inline();
```

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by dean-long — APPROVED** [review]

**Comment by iklam** [comment]
Thanks @dean-long and @vnkozlov for the review
/integrate

**Pull request closed without merging**